test/e2e/pods.go

```diff
@@ -555,6 +578,233 @@ var _ = Describe("Pods", func() {
 		}, false)
 	})
 
+	It("should reset back-off on image update", func() {
```

Comment:
I think we can word this more clearly.  Remember that "Pods" will be prepended and the results should be a sentence that describes exactly what the test is testing.  Perhaps something like
"should have their auto-restart back-off timer reset on image update"?